src/violations.rs

```diff
@@ -1157,6 +1157,73 @@ impl Violation for ConsiderUsingFromImport {
     }
 }
 
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum ViolationsCmpop {
+    Eq,
+    NotEq,
+    Lt,
+    LtE,
+    Gt,
+    GtE,
+    Is,
+    IsNot,
+    In,
+    NotIn,
+}
+
+impl From<&Cmpop> for ViolationsCmpop {
+    fn from(cmpop: &Cmpop) -> Self {
+        match cmpop {
+            Cmpop::Eq => Self::Eq,
+            Cmpop::NotEq => Self::NotEq,
+            Cmpop::Lt => Self::Lt,
+            Cmpop::LtE => Self::LtE,
+            Cmpop::Gt => Self::Gt,
+            Cmpop::GtE => Self::GtE,
+            Cmpop::Is => Self::Is,
+            Cmpop::IsNot => Self::IsNot,
+            Cmpop::In => Self::In,
+            Cmpop::NotIn => Self::NotIn,
+        }
+    }
+}
+
+impl fmt::Display for ViolationsCmpop {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let representation = match self {
+            Self::Eq => "==",
+            Self::NotEq => "!=",
+            Self::Lt => "<",
+            Self::LtE => "<=",
+            Self::Gt => ">",
+            Self::GtE => ">=",
+            Self::Is => "is",
+            Self::IsNot => "is not",
+            Self::In => "in",
+            Self::NotIn => "not in",
+        };
+        write!(f, "{representation}")
+    }
+}
+
+define_violation!(
+    pub struct ConstantComparison(pub String, pub String, pub ViolationsCmpop);
```

Comment:
I know that most violation structs currently use tuple structs but we actually want to change all these to field structs in the future whenever the types don't clearly identify the purpose of the field. So in this case I think the following would be preferable:

```rs
pub struct ConstantComparison {
    pub left_const: String,
    pub op: ViolationsCmpop,
    pub right_const: String,
}
```